docs/hacking.rst

```diff
@@ -0,0 +1,212 @@
+Hacking on :mod:`zope.meta`
+===========================
+
+
+Getting the Code
+################
+
+The main repository for :mod:`zope.meta` is in the Zope Foundation
+Github repository:
+
+  https://github.com/zopefoundation/zope.meta
+
+You can get a read-only checkout from there:
+
+.. code-block:: sh
+
+   $ git clone https://github.com/zopefoundation/zope.meta.git
+
+or fork it and get a writeable checkout of your fork:
+
+.. code-block:: sh
+
+   $ git clone git@github.com/jrandom/zope.meta.git
+
+
+Working in a Python virtual environment
+#######################################
+
+Installing
+----------
+
+You can use Python's standard ``venv`` package to create lightweight Python
+development environments, where you can run the tests using nothing more
+than the ``python`` binary in a virtualenv.  First, create a scratch
+environment:
+
+.. code-block:: sh
+
+   $ python3.12 -m venv /tmp/hack-zope.meta
+
+Next, install this package in "development mod" in the newly-created
+environment:
+
+.. code-block:: sh
+
+   $ /tmp/hack-zope.meta/bin/pip install -e .
+
+Running the tests
+-----------------
+
+You can install test tools using the ``test`` extra:
+
+.. code-block:: sh
+
+   $ /tmp/hack-zope.meta/bin/pip install -e ".[test]"
+
+
+That command installs the tools needed to run
+the tests:  in particular, the ``zope.testrunner`` (see
+:external+testrunner:std:doc:`getting-started`) and
+:external+coverage:std:doc:`index` tools.
+
+To run the tests via ``zope.testrunner``:
+
+.. code-block:: sh
+
+   $ /tmp/hack-zope.meta/bin/zope-testrunner --test-path=src
+   Running zope.testrunner.layer.UnitTests tests:
+     Set up zope.testrunner.layer.UnitTests in 0.000 seconds.
+     Ran 41 tests with 0 failures, 0 errors and 0 skipped in 0.003 seconds.
+   Tearing down left over layers:
+     Tear down zope.testrunner.layer.UnitTests in 0.000 seconds.
```

Comment:
Do we have tests for this code now?  That is wonderful news!

Reply:
No, it is sample output, not reality, and it's gone now.